### DISCUSSION TITLE
P2 Repository for usage within Eclispe target files?

### DISCUSSION THREAD

**Discussion by rubenporras** [TNG/ArchUnit#1018] [General]
Hi,

is there a P2 Repository for usage within Eclispe target files? I could not find any so far.

If not, is there any plans of providing one?

Kind Regards

**Comment by codecholeric** [comment]
Sorry, there are no current plans for such a repository. What are your reasons for not simply using Maven Central?

**Reply by rubenporras** [reply]
Hi,

that Maven Central artifacts cannot be used in eclipse target files, so teams using eclipse target files (what is the normal case for eclipse plugin development afaik), cannot use AnchUnit, and that is a pity.

**Reply by codecholeric** [reply]
My problem is that I have no experience whatsoever with P2 repositories, so I'm afraid it's out of scope for me, since I don't have the time to research everything necessary and then maintain this in the long run :disappointed: 
I could support somebody with knowledge to do it though.
Other than that, since ArchUnit is only interested in compiled files and doesn't care about OSGI bundles, etc., it would also be possible to set up a non-OSGI project and simply specify the file path of the compiled OSGI classes as input.
